lib/wrapperService.ts

```diff
@@ -0,0 +1,175 @@
+   * `wrappers` and `wrappersByType` properties.
+   */
+  public static factories: ArcWrapperFactories = {
+    AbsoluteVote: AbsoluteVote as ContractWrapperFactory<AbsoluteVoteWrapper>,
+    ContributionReward: ContributionReward as ContractWrapperFactory<ContributionRewardWrapper>,
+    DaoCreator: DaoCreator as ContractWrapperFactory<DaoCreatorWrapper>,
+    GenesisProtocol: GenesisProtocol as ContractWrapperFactory<GenesisProtocolWrapper>,
+    GlobalConstraintRegistrar: GlobalConstraintRegistrar as ContractWrapperFactory<GlobalConstraintRegistrarWrapper>,
+    SchemeRegistrar: SchemeRegistrar as ContractWrapperFactory<SchemeRegistrarWrapper>,
+    TokenCapGC: TokenCapGC as ContractWrapperFactory<TokenCapGCWrapper>,
+    UpgradeScheme: UpgradeScheme as ContractWrapperFactory<UpgradeSchemeWrapper>,
+    VestingScheme: VestingScheme as ContractWrapperFactory<VestingSchemeWrapper>,
+    VoteInOrganizationScheme: VoteInOrganizationScheme as ContractWrapperFactory<VoteInOrganizationSchemeWrapper>,
+  };
+
+  /**
+   * initialize() must be called before any of the static properties will have values.
+   * It is currently called in ArcInitialize(), which in trun must be invoked by any applicaiton
+   * using Arc.js.
+   */
+  public static async initialize(): Promise<void> {
+    LoggingService.debug("WrapperService: initializing");
+    /**
+     * Deployed contract wrappers by name.
+     */
+    WrapperService.wrappers = {
+      AbsoluteVote: await AbsoluteVote.deployed(),
+      ContributionReward: await ContributionReward.deployed(),
+      DaoCreator: await DaoCreator.deployed(),
+      GenesisProtocol: await GenesisProtocol.deployed(),
+      GlobalConstraintRegistrar: await GlobalConstraintRegistrar.deployed(),
+      SchemeRegistrar: await SchemeRegistrar.deployed(),
+      TokenCapGC: await TokenCapGC.deployed(),
+      UpgradeScheme: await UpgradeScheme.deployed(),
+      VestingScheme: await VestingScheme.deployed(),
+      VoteInOrganizationScheme: await VoteInOrganizationScheme.deployed(),
+    };
+
+    /**
+     * Contract wrappers grouped by type
+     */
+    WrapperService.wrappersByType = {
+      allWrappers: WrapperService.wrappers,
+      globalConstraints: [
+        WrapperService.wrappers.TokenCapGC,
+      ],
+      other: [
+        WrapperService.wrappers.DaoCreator,
+      ],
+      schemes: [
+        WrapperService.wrappers.ContributionReward,
+        WrapperService.wrappers.GenesisProtocol,
+        WrapperService.wrappers.GlobalConstraintRegistrar,
+        WrapperService.wrappers.SchemeRegistrar,
+        WrapperService.wrappers.UpgradeScheme,
+        WrapperService.wrappers.VestingScheme,
+        WrapperService.wrappers.VoteInOrganizationScheme,
+      ],
+      votingMachines: [
+        WrapperService.wrappers.AbsoluteVote,
+        WrapperService.wrappers.GenesisProtocol,
+      ],
+    };
+  }
+
+  /**
+   * Returns an Arc.js contract wrapper or undefined if not found.
```

Comment:
actually it returns a promise no?

Reply:
yeah, I tend not too mention that it is a promise in those comments.  maybe I should.